### ISSUE TITLE
2 versions of tensorboard get installed with SCT

### ISSUE THREAD

**Issue by joshuacwnewton** [neuropoly/spinalcordtoolbox#3035]
When SCT is installed, 2 versions of tensorboard get installed:

* ~~`tensorboard==2.3.0`~~ `tensorboard==2.4.0` (i.e. the latest version), which gets installed [because of `ivadomed`](https://github.com/ivadomed/ivadomed/blob/master/requirements.txt#L13).
* `tensorflow-tensorboard==1.5.1`, which is ["a relic of a bygone era; there’s no need to install it anymore."](https://github.com/tensorflow/tensorboard/issues/1862#issuecomment-521876133)
    * The reason the older, differently-named `tensorflow-tensorboard` gets installed is because it's a dependency for `tensorflow==1.5.0`. (The newer-named `tensorboard` was only [introduced in Tensorflow 1.6.0+](https://pypi.org/project/tensorflow-tensorboard/).)

**This has the potential to create conflicts.** For example, it is currently roadblocking #3034. An unrelated import from `ivadomed` triggers a new exception [in this CI build](https://travis-ci.org/github/neuropoly/spinalcordtoolbox/jobs/743004132#L2003):

```
unit_testing/test_deepseg.py:16: in <module>
    import sct_deepseg
scripts/sct_deepseg.py:18: in <module>
    from ivadomed import inference as imed_inference
python/envs/venv_sct/lib/python3.6/site-packages/ivadomed/inference.py:16: in <module>
    from ivadomed import training as imed_training
python/envs/venv_sct/lib/python3.6/site-packages/ivadomed/training.py:13: in <module>
    from torch.utils.tensorboard import SummaryWriter
python/envs/venv_sct/lib/python3.6/site-packages/torch/utils/tensorboard/__init__.py:2: in <module>
    from tensorboard.summary.writer.record_writer import RecordWriter  # noqa F401
python/envs/venv_sct/lib/python3.6/site-packages/tensorboard/summary/__init__.py:31: in <module>
    from tensorboard.summary import v2
python/envs/venv_sct/lib/python3.6/site-packages/tensorboard/summary/v2.py:24: in <module>
    from tensorboard.plugins.audio.summary_v2 import audio
python/envs/venv_sct/lib/python3.6/site-packages/tensorboard/plugins/audio/summary_v2.py:30: in <module>
    from tensorboard.compat import tf2 as tf
python/envs/venv_sct/lib/python3.6/site-packages/tensorboard/compat/__init__.py:28: in <module>
    import tensorboard.lazy as _lazy
E   AttributeError: module 'tensorboard' has no attribute 'lazy'
```

This exception is specifically [because 2 versions are installed](https://github.com/tensorflow/tensorboard/issues/1862#issuecomment-521874990). Ideally, we should use just `tensorboard`, but our `tensorflow==1.5.0` requirement is preventing that.

**Comment by joshuacwnewton** [comment]
For context, the reason why we are forced to use `tensorflow==1.5.0` is documented in #2618.

**Comment by jcohenadad** [comment]
thank you for catching this @joshuacwnewton 🙏

**Comment by lrouhier** [comment]
What if we 'uninstalled' the `tensorflow-tensorboard` package before installing the `tensorboard` package so that this is not a problem between now and the time we finish to move to pytorch?

**Comment by joshuacwnewton** [comment]
> What if we 'uninstalled' the `tensorflow-tensorboard` package before installing the `tensorboard` package so that this is not a problem between now and the time we finish to move to pytorch?

It depends on what `tensorflow==1.5.0` needs to function properly. It's possible that for our (minimal) usage of tensorflow, `tensorflow-tensorboard` isn't required. Manually interfering with dependencies like that does feel a little hacky, though. 

But, that idea seems quick enough to try out in #3034 (just to see how CI behaves with that in place).

**Comment by joshuacwnewton** [comment]
Two things:
* The fix from #3034 was included into #3041, so in theory this issue (#3035) should be closed
* However, the fix is included in this "else" block, i.e. just for the dev installation, so it won't get applied for the release installation. (So, the release installation will fail if this isn't fixed!)

https://github.com/neuropoly/spinalcordtoolbox/blob/dc97d42730d7e603f83e54c68f242ad2bef7189b/install_sct#L531-L543

I'll open a PR to address this.

**Comment by joshuacwnewton** [comment]
I also notice that the pip status check immediately afterwards now no longer works as intended, as it will only verify `pip install tensorboard` and none of the preceding commands:

https://github.com/neuropoly/spinalcordtoolbox/blob/dc97d42730d7e603f83e54c68f242ad2bef7189b/install_sct#L537-L548
